src/Action/UserFlairAction.ts

```diff
@@ -28,51 +24,49 @@ export class UserFlairAction extends Action {
     const dryRun = runtimeDryrun || this.dryRun;
     let flairParts = [];
 
-    if (this.flair_template_id !== '') {
+    if (this.flair_template_id !== undefined) {
       flairParts.push(`Flair template ID: ${this.flair_template_id}`)
+      if(this.text !== undefined || this.css !== undefined) {
+        this.logger.warn('Text/CSS properties will be ignored since a flair template is specified');
+      }
     } else {
-      if (this.text !== '') {
+      if (this.text !== undefined) {
         flairParts.push(`Text: ${this.text}`);
       }
-      if (this.css !== '') {
+      if (this.css !== undefined) {
         flairParts.push(`CSS: ${this.css}`);
       }
     }
 
-    const flairSummary = flairParts.length === 0 ? 'No user flair (unflaired)' : flairParts.join(' | ');
+    const flairSummary = flairParts.length === 0 ? 'Unflair user' : flairParts.join(' | ');
     this.logger.verbose(flairSummary);
-    if (item && item?.author) {
-      if (!this.dryRun) {
-        if (this.flair_template_id !== '') {
+
+    if (!this.dryRun) {
+      if (this.flair_template_id !== undefined) {
+        try {
           // @ts-ignore
           await this.client.assignUserFlairByTemplateId({
             subredditName: item.subreddit.display_name,
             flairTemplateId: this.flair_template_id,
             username: item.author.name,
-          })
-            .then((e: any) => {
-              this.logger.verbose(JSON.stringify(e));
-            })
-            .catch((e: any) => {
-              this.logger.verbose(JSON.stringify(e));
-            })
+          });
+        } catch (err: any) {
+          this.logger.error('Either the flair template ID is incorrect or you do not have permission to access it.');
+          throw err;
         }
+      } else if (this.text === undefined && this.css === undefined) {
```

Comment:
I modified the config properties so if there are none specified or all are empty then it deletes the user flair. Easy functionality gain.

Reply:
Cool! That's useful and simple solution!